### PR TITLE
Upgrade to automation api v0.1.0

### LINKED ISSUE #15: Clear labels and envs_vars after app is created

### LINKED ISSUE #16: Support new API Model
https://github.com/rht-labs/api-design

### LINKED ISSUE #17: Support Adding Users to A Project
So projects can be shared amongst a team per the new api model

### LINKED ISSUE #18: Support Multiple Clusters in One Execution

### LINKED ISSUE #19: Investigate Why Ansible Is Resolving Variables on Skipped Tasks
This is producing warnings that are ominous like the below:

```
TASK [create-openshift-resources : include] ************************************
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this will be a fatal error.: 'openshift_resources' 
is undefined.
This feature will be removed in a future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
```

### PR DESCRIPTION
**Pull request by sherl0cks**
#### What does this PR do?

Introduces support for the `project.user_to_role` construct in the automation API. Under the covers, this maps to `oadm policy add-role-to-user.` The major change architectural is that this role now needs to run on an openshift master in order to get access to the `oadm` cli. Previously, this role could run anywhere the oc binary was.
#### Which tests illustrate how this code works?

The test framework has been updated to use the CDK. See [the create-openshift-resources docs](roles/create-openshift-resources/README.md) to get that configured. The test `cdk-create-openshift-resources.yml` illustrates the functionality once you have the CDK working.
#### Is there a relevant Issue open for this?
- resolves #19 
- resolves #18 
- resolves #17 
- resolves #16 
- resolves #15
#### Are there any other relevant resources that should be reviewed as well?

Some of the odd stuff in ansible has been answered:
- http://docs.ansible.com/ansible/playbooks_roles.html#dynamic-versus-static-includes
- http://docs.ansible.com/ansible/playbooks_loops.html#loops-and-includes-in-2-0
#### Who would you like to review this?

/cc @oybed

### CODE REVIEW

File: roles/create-openshift-resources/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Create OpenShift Resources Role
+
+An ansible role that consumes the model defined in the [Open Innovation Labs Automation API](https://github.com/rht-labs/api-design). The API declares a desired state in one or more OpenShift clusters and this role is responsible for creating that state. Both projects are in early stages, so unless otherwise noted, this work is being developed against the master branch of the Atuomation API. Future versions will look to stabilize against a tagged release of the API.
```
Comment (oybed) [inline roles/create-openshift-resources/README.md]: typo: "Atuomation"

File: roles/create-openshift-resources/tasks/configure_openshift_cluster.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  when: openshift_resources_present == true and openshift_password is not defined or openshift_password == ''
+
+- name: "Fail for Missing OpenShift Host Env"
+  fail: msg="This role requires openshift_host_env to be set and non empty"
+  when: openshift_resources_present == true and openshift_host_env is not defined or openshift_host_env == ""
+
+- name: "Log in to OpenShift Client"
+  command: >
+    {{ openshift.common.client_binary }} login {{ openshift_host_env }}
+    --insecure-skip-tls-verify=true --username={{ openshift_user }} --password={{ openshift_password }}
```
Comment (oybed) [inline roles/create-openshift-resources/tasks/configure_openshift_cluster.yml:30]: Skipping the cert may be ok for now, but we really should come up with a solution where the cert is used.

File: roles/create-openshift-resources/tasks/create_app.yml [hunk 3]
```diff
@@ -15,21 +20,23 @@
...
+    app_source: "{{ app.base_image }}~{{ app.scm_url }}"
+  when: app.scm_url is defined and app.scm_url != '' and app.base_image is defined and app.base_image != ''
 
 - name: "Set App Source with only Base Image"
   set_fact:
     app_options: "{{ app_options }} --image-stream={{ app.base_image }}"
-  when: (app.base_image is defined and app.base_image != '') and (app.git_url is not defined or app.git_url == '')
+  when: (app.base_image is defined and app.base_image != '') and (app.scm_url is not defined or app.scm_url == '')
 
 - name: "Set App Source with only Git Url"
```
Comment (oybed) [inline roles/create-openshift-resources/tasks/create_app.yml]: Please update from "Git Url" to "SCM URL" (or something like that)

File: roles/create-openshift-resources/tasks/create_app.yml [hunk 4]
```diff
@@ -88,8 +103,8 @@
   delay: 5
   ignore_errors: yes
 
-## This is confusing to me. Dunno why the app_item is not passed to the includes context...
-- name: Set Facts Not Sure Why
+# http://docs.ansible.com/ansible/playbooks_loops.html#loops-and-includes-in-2-0
+- name: Set Facts
```
Comment (oybed) [inline roles/create-openshift-resources/tasks/create_app.yml]: Personally I'd prefer to see a bit more specific name than just "Set Facts" - makes it harder to troubleshoot in the event of a problem with very generic names.

**Comment by oybed** [comment]
@sherl0cks reviewed the changes - a few minor comments, but overall looks good.

File: roles/create-openshift-resources/tasks/configure_openshift_cluster.yml (hunk 2 above)
Comment (sherl0cks) [reply roles/create-openshift-resources/tasks/configure_openshift_cluster.yml:30]: https://github.com/rht-labs/ansible-stacks/issues/22

File: roles/create-openshift-resources/tasks/create_app.yml (hunk 3 above)
Comment (sherl0cks) [reply roles/create-openshift-resources/tasks/create_app.yml]: That's precisely what this commit does. Your comment is on source that was removed in the commit.

File: roles/create-openshift-resources/README.md (hunk 1 above)
Comment (sherl0cks) [reply roles/create-openshift-resources/README.md]: Fixed

File: roles/create-openshift-resources/tasks/create_app.yml (hunk 4 above)
Comment (sherl0cks) [reply roles/create-openshift-resources/tasks/create_app.yml]: Fixed. Please do note the comment with link to documentation above the task as well

**Comment by sherl0cks** [comment]
@oybed fixed

File: roles/create-openshift-resources/tasks/create_app.yml (hunk 3 above)
Comment (oybed) [reply roles/create-openshift-resources/tasks/create_app.yml]: @sherl0cks no - my comment is for line 26/36 - e.g: the "- name: "Set App Source with only Git Url".

Comment (sherl0cks) [reply roles/create-openshift-resources/tasks/create_app.yml]: @oybed fixed

**Pull request merged**